plugins/nomad-backend/src/service/router.ts

```diff
@@ -0,0 +1,132 @@
+import { InputError } from '@backstage/errors';
+import express from 'express';
+import Router from 'express-promise-router';
+import { Logger } from 'winston';
+
+/** @public */
+export interface RouterOptions {
+  logger: Logger;
+  config: Config;
+}
+
+/** @public */
+export async function createRouter(
+  options: RouterOptions,
+): Promise<express.Router> {
+  const { config, logger } = options;
+
+  // Get Nomad addr and token from config
+  const addr = config.getString('nomad.addr');
+  const token = config.getOptionalString('nomad.token');
+
+  const router = Router();
+  router.use(express.json());
+
+  router.get('/health', (_, resp) => {
+    resp.json({ status: 'ok' });
+  });
+
+  router.get('/v1/allocations', async (req, res) => {
+    // Check namespace argument
+    const namespace = (req.query.namespace as string) ?? '';
+    if (!namespace || namespace === '') {
+      throw new InputError(`Missing "namespace" query parameter`);
+    }
+
+    // Check filter argument
+    const filter = (req.query.filter as string) ?? '';
+    logger.debug(`request headers: namespace=${namespace} filter=${filter}`);
+
+    // Issue the request
+    const endpoint = `${addr}/v1/allocations?namespace=${encodeURIComponent(
+      namespace,
+    )}&filter=${encodeURIComponent(filter)}`;
+    const allocationsResp = await fetch(endpoint, {
+      method: 'GET',
+      headers: {
+        Accept: 'application/json',
+        'X-Nomad-Token': token || '',
+      },
+    });
+
+    // Check response
+    if (allocationsResp.status !== 200) {
+      const body = await allocationsResp.text();
+      logger.error(`failed to call /v1/allocations endpoint: ${body}`);
+      res.status(allocationsResp.status).json({
+        message: body,
+      });
+      return;
+    }
+
+    // Deserialize and return
+    const allocationsBody = await allocationsResp.json();
+    logger.debug(`/v1/allocations response: ${allocationsBody}`);
+    res.json(allocationsBody);
+  });
+
+  router.get('/v1/job/:job_id/versions', async (req, resp) => {
+    // Check namespace argument
+    const namespace = (req.query.namespace as string) ?? '';
+    if (!namespace || namespace === '') {
+      throw new InputError(`Missing "namespace" query parameter`);
+    }
+
+    // Get job ID
+    const jobID = (req.params.job_id as string) ?? '';
+
+    logger.info(`token: ${token}`);
+
+    // Issue the request
+    const apiResp = await fetch(
+      `${addr}/v1/job/${jobID}/versions?namespace=${encodeURIComponent(
+        namespace,
+      )}`,
+      {
+        method: 'GET',
+        headers: {
+          Accept: 'application/json',
+          'X-Nomad-Token': token || '',
+        },
+      },
+    );
+
+    // Check response
+    if (apiResp.status !== 200) {
+      const body = await apiResp.text();
+      logger.error(`failed to call /v1/job/:job_id/versions endpoint: ${body}`);
+      resp.status(apiResp.status).send({
+        message: body,
+      });
```

Comment:
```suggestion
      res.status(apiResp.status).json({message: body})
```

Let's replace this usage of send as well :)